snow/engine/avalanche/transitive_test.go

```diff
@@ -3500,3 +3502,145 @@ func TestEngineDoubleChit(t *testing.T) {
+		ids.GenerateTestID(),
+		ids.GenerateTestID(),
+	}
+
+	tx0 := &snowstorm.TestTx{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		InputIDsV: utxos[:1],
+	}
+	tx1 := &snowstorm.TestTx{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		InputIDsV: utxos[1:2],
+	}
+	tx2 := &snowstorm.TestTx{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		InputIDsV: utxos[1:2],
+	}
+
+	vtx := &avalanche.TestVertex{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		HeightV: 0,
+		TxsV:    []snowstorm.Tx{tx0},
+		BytesV:  []byte{0},
+	}
+
+	missingVtx := &avalanche.TestVertex{TestDecidable: choices.TestDecidable{
+		IDV:     ids.GenerateTestID(),
+		StatusV: choices.Unknown,
+	}}
+
+	pendingVtx0 := &avalanche.TestVertex{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		ParentsV: []avalanche.Vertex{vtx, missingVtx},
+		HeightV:  1,
+		TxsV:     []snowstorm.Tx{tx1},
+		BytesV:   []byte{1},
+	}
+
+	pendingVtx1 := &avalanche.TestVertex{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		ParentsV: []avalanche.Vertex{pendingVtx0},
+		HeightV:  2,
+		TxsV:     []snowstorm.Tx{tx2},
+		BytesV:   []byte{2},
+	}
+
+	manager.EdgeF = func() []ids.ID { return nil }
+	manager.GetF = func(id ids.ID) (avalanche.Vertex, error) {
+		switch id {
+		case vtx.ID():
+			return vtx, nil
+		case missingVtx.ID():
+			return nil, errMissing
+		case pendingVtx0.ID():
+			return pendingVtx0, nil
+		case pendingVtx1.ID():
+			return pendingVtx1, nil
+		}
+		assert.FailNow(t, "unknown vertex", "vtxID: %s", id)
+		panic("should have errored")
+	}
+
+	te := &Transitive{}
+	err = te.Initialize(config)
+	assert.NoError(t, err)
+
+	queryReqID := new(uint32)
+	queried := new(bool)
+	sender.PushQueryF = func(inVdrs ids.ShortSet, requestID uint32, vtxID ids.ID, _ []byte) {
+		assert.Len(t, inVdrs, 1, "wrong number of validators")
+		*queryReqID = requestID
+		assert.Equal(t, vtx.ID(), vtxID, "wrong vertex requested")
+		*queried = true
+	}
+
+	getReqID := new(uint32)
+	fetched := new(bool)
+	sender.GetF = func(inVdr ids.ShortID, requestID uint32, vtxID ids.ID) {
+		assert.Equal(t, vdr, inVdr, "wrong validator")
+		*getReqID = requestID
+		assert.Equal(t, missingVtx.ID(), vtxID, "wrong vertex requested")
+		*fetched = true
+	}
+
+	issued, err := te.issueFrom(vdr, pendingVtx1)
+	assert.NoError(t, err)
+	assert.False(t, issued, "shouldn't have been able to issue %s", pendingVtx1.ID())
+	assert.True(t, *queried, "should have queried for %s", vtx.ID())
+	assert.True(t, *fetched, "should have fetched %s", missingVtx.ID())
+
+	err = te.Chits(vdr, *queryReqID, []ids.ID{pendingVtx1.ID()})
```

Comment:
// Can't apply votes yet because pendingVtx0 isn't issued because missingVtx is missing